addons/website_sale_coupon/wizard/coupon_share.py

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+
+from odoo import fields, models, api
+
+
+class CouponShare(models.TransientModel):
+    _name = 'coupon.share.wizard'
+    _description = 'Share Coupon'
+
+    coupon_id = fields.Many2one(comodel_name='coupon.coupon', required=True)
+    program_website_id = fields.Many2one(comodel_name='website', related='coupon_id.program_id.website_id')
```

Comment:
You can omit `comodel_name` (unless you really care about it but we never explicitely write it, actually it is the first time I see it written like that :yum: )